eng/intellisense.targets

```diff
@@ -0,0 +1,41 @@
+<Project>
+
+  <PropertyGroup>
+    <XmlDocFileRoot>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', 'microsoft.private.intellisense', '$(MicrosoftPrivateIntellisenseVersion)', 'IntellisenseFiles'))</XmlDocFileRoot>
+  </PropertyGroup>
+
+  <!-- If the project is not generating its own documentation yet, retrieve it from the dotnet-api-docs nuget package. -->
+  <Choose>
+    <When Condition="('$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false') and Exists($([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'net', '1033', '$(AssemblyName).xml')))">
```

Comment:
Unfortunately "Choose / When" constructs are quite verbose in msbuild syntax which is why they are hardly used.
You don't gain anything by using them here as you can achieve the same with a far less verbose syntax. Put the `"'$(UseIntellisenseDocumentationFile)' != 'true'"` condition on the property group and then use the `Exists` function on the two property conditions inside it.

Reply:
IMO we don't even need `UseIntellisenseDocumentationFile` to be considered in these properties.  Setting the value for IntellisenseFilePath can be done unconditionally.  We might even just move it inside ChangeDocumentationFile... target since it's only ever used there.

Reply:
> IMO we don't even need UseIntellisenseDocumentationFile to be considered in these properties. Setting the value for IntellisenseFilePath can be done unconditionally

We only want to use the Exists msbuild function when necessary as it results in an underlying IO call. Setting these properties unconditionally means that we always have at least one IO call vs. when conditioning on `UseIntellisenseDocumentationFile`, we would avoid that when the intellisense package file isn't used.

> We might even just move it inside ChangeDocumentationFile... target since it's only ever used there.

... which makes the Exists msbuild function conditional and that's what we want. Either inside or outside the target works for me. It boils down to if we want to do the Exists check lazily or during evaluation time. Lazily is probably better to avoid unnecessary work during design time and restore builds.